src/helper/forward.h

```diff
@@ -24,6 +26,9 @@ class Request;
 
 } // namespace Helper
 
+using HelperPointer = RefCount<helper>;
+using StatefulHelperPointer = RefCount<statefulhelper>;
```

Comment:
If you keep these new types, please move into the ```namespace Helper``` scope.

Reply:
IMO, the primary reason we should not move these aliases is because such a move would create a "Foo inside Foo" problem: Helper::HelperPointer does not look good and Helper::Pointer is just wrong. The problem here is in the name of the `Helper` namespace or the name of the `::helper` class (that this PR does not and should not change).

I hope we will rename the `::helper` class eventually. If we can agree on that future class name _now,_ then we can move these new pointers inside the Helper namespace using that new name in _this_ PR (and then rename the classes in another dedicated PR).

I can suggest Helper::Client (for the current `::helper` class) and Helper::Session (for the current `helper_server`). That naming scheme matches what the corresponding classes represent well. Squid already uses similar Client terminology for HTTP and ICAP clients.

That (future) naming scheme would result in _this_ PR using Helper::ClientPointer and Helper::StatefulClientPointer. Another dedicated PR will then rename the other/existing classes and move them into the Helper namespace (and into the src/helper/ directory?). We just need to merge a few existing bug fixes (like this PR) so it should not take long.

@yadij, would you support this new Client/Session naming scheme for helper-related classes?

Reply:
I am okay with that naming. Yes.

Reply:
Done (72c531e).